### PR TITLE
ACS-2827: Add Search Tests To CMIS-TAS

### PR DESCRIPTION
**Pull request by KikoUK**

### CODE REVIEW

File: packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractCmisE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @AfterMethod(alwaysRun = true)
+    public void showEndTestInfo(Method method)
+    {
+        LOGGER.info(String.format("*** ENDING Test: [%s] ***", method.getName()));
+    }
+
+    public Integer getElasticWaitTimeInSeconds()
+    {
+        return cmisProperties.envProperty().getSolrWaitTimeInSeconds();
```
[the remark this answers is not in the document]
Comment (KikoUK) [reply packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractCmisE2ETest.java:42]: This is one of the parts I need to change once this PR is merged in https://github.com/Alfresco/alfresco-tas-utility as currently there is no functionality related to ES in most of the tas-utility/api projects.

File: packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/ElasticSearchInFolderTests.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.FileType;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.QueryModel;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class ElasticSearchInFolderTests extends AbstractCmisE2ETest
```
[the remark this answers is not in the document]
Comment (KikoUK) [reply packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/ElasticSearchInFolderTests.java:15]: These tests will be ran in acs-packaging through the functionality added in this [PR](https://github.com/Alfresco/acs-packaging/pull/2230)

We can also run them against solr in community-repo if needed as there are already some solr related tests in community-repo.

**Review by dsibilio — APPROVED** [review]

File: packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+		<query value="SELECT * FROM cmis:document where IN_FOLDER('%s') ORDER BY cmis:createdBy DESC" expectedResults="5" />
+		<query value="SELECT * FROM cmis:document where IN_FOLDER('%s') AND cmis:name IS NOT NULL" expectedResults="5" />
+		<query value="SELECT * FROM cmis:folder where IN_FOLDER('%s') AND cmis:name IS NOT NULL" expectedResults="3" />
+		<query value="SELECT * FROM cmis:document where IN_FOLDER('%s') AND cmis:name LIKE 'fourthFile'" expectedResults="1" />
+		<query value="SELECT * FROM cmis:folder where IN_FOLDER('%s') AND NOT(cmis:name NOT IN ('subFolder'))" expectedResults="1" />
+		<query value="SELECT * FROM cmis:document where IN_FOLDER('%s') AND cmis:name IN ('fourthFile', 'fifthFile.txt')" expectedResults="2" />
+		<query value="SELECT * FROM cmis:document where IN_FOLDER('%s') AND cmis:name NOT IN ('fourthFile', 'fifthFile.txt')" expectedResults="3" />
+		<query value="SELECT * FROM cmis:folder where IN_FOLDER('%s') AND cmis:name &lt;&gt; 'subFolder'" expectedResults="2" />
+		<query value="SELECT cmis:secondaryObjectTypeIds FROM cmis:folder where IN_FOLDER('%s') AND cmis:name = 'subFolder'" expectedResults="1" />
```
Comment (tpage-alfresco) [inline packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml:21]: I dislike this style of test. The error messages will be something like "Expected 1 got 2" and historically people have "fixed" this sort of issue by simply changing 1 to 2 in the test.

File: packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractE2EFunctionalTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+
+@ContextConfiguration ("classpath:alfresco-cmis-context.xml")
+public abstract class AbstractE2EFunctionalTest extends AbstractTestNGSpringContextTests
+{
+    /** The number of retries that a query will be tried before giving up. */
+    protected static final int SEARCH_MAX_ATTEMPTS = 1;
```
Comment (tpage-alfresco) [inline packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractE2EFunctionalTest.java:21]: Why only a single attempt?

Comment (KikoUK) [reply packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractE2EFunctionalTest.java:21]: So the builds don't take too long. 

How many attempts would be ideal?

File: packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml (hunk 3 above)
Comment (KikoUK) [reply packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml:21]: In this [PR](tests/tas-cmis/src/test/resources/cmis-elastic-suite.xml)

Comment (montgolfiere) [reply packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml:21]: I'm not sure PR link is correct ?

File: packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractE2EFunctionalTest.java (hunk 4 above)
Comment (montgolfiere) [reply packaging/tests/tas-cmis/src/test/java/org/alfresco/cmis/search/AbstractE2EFunctionalTest.java:21]: For comparison, see also: https://github.com/Alfresco/acs-packaging/blob/master/tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/SearchQueryService.java#L80 (retries every 1 sec for up to 20 sec) ?

File: packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml (hunk 3 above)
Comment (KikoUK) [reply packaging/tests/tas-cmis/src/test/resources/search-in-folder.xml:21]: I'm not sure if we're thinking of the same thing, but the data provider comes from this line: https://github.com/Alfresco/alfresco-community-repo/pull/1125/files#diff-b134b8e722005ac2473b2dab6009d070a5ef36c84790bda11b7d3c7ec404d8b0R54
and then the getQueries comes from here: https://github.com/Alfresco/alfresco-tas-utility/blob/master/src/main/java/org/alfresco/utility/data/provider/XMLTestDataProvider.java

Or do you mean the test data needed for the tests which is created in the @BeforeClass here https://github.com/Alfresco/alfresco-community-repo/pull/1125/files#diff-b134b8e722005ac2473b2dab6009d070a5ef36c84790bda11b7d3c7ec404d8b0R21

**Pull request merged**